From 628b9428521396a5ce9ad96d9e485bbe52d84a71 Mon Sep 17 00:00:00 2001
From: Samantha N. Bueno <sbueno+anaconda@redhat.com>
Date: Mon, 06 May 2013 18:41:48 +0000
Subject: Force a password to be set if option checked in TUI. (#927956)

Require a password either for root or the created admin user.
---
diff --git a/pyanaconda/ui/tui/spokes/user.py b/pyanaconda/ui/tui/spokes/user.py
index a6d36cb..ce53e1b 100644
--- a/pyanaconda/ui/tui/spokes/user.py
+++ b/pyanaconda/ui/tui/spokes/user.py
@@ -65,16 +65,26 @@ class UserSpoke(FirstbootSpokeMixIn, EditTUISpoke):
 
     @property
     def completed(self):
-        return len(self.data.user.userList) > 0
+        """ Verify a user is created; verify pw is set if option checked. """
+        if len(self.data.user.userList) > 0:
+            if self.args._use_password and not bool(self.args.password or self.args.isCrypted):
+                return False
+            else:
+                return True
+        else:
+            return False
 
     @property
     def mandatory(self):
-        return (not self.data.rootpw.password) or self.data.rootpw.lock
+        """ Only mandatory if root account is disabled. """
+        return not bool(self.data.rootpw.password) or self.data.rootpw.lock
 
     @property
     def status(self):
         if len(self.data.user.userList) == 0:
             return _("No user will be created")
+        elif self.args._use_password and not bool(self.args.password or self.args.isCrypted):
+            return _("You must set a password")
         elif "wheel" in self.data.user.userList[0].groups:
             return _("Administrator %s will be created") % self.data.user.userList[0].name
         else:
--
cgit v0.9.1
